Bottom: 4f107a7259394aab872708271a086ea950cc262e
Top:    3425768aeef91280ffe856fcdf1ebab4c0c24952
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-24 17:44:49 +0100

TEMP


---

diff --git a/src/draw.c b/src/draw.c
index f12020d..d096377 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -102,6 +102,7 @@ static void DrawEMark (ElementTypePtr, LocationType, LocationType, bool);
 static void ClearPad (PadTypePtr, bool);
 static void DrawHole (PinTypePtr);
 static void DrawMask (int side, BoxType *);
+static void DrawPaste (int side, BoxType *);
 static void DrawRats (BoxType *);
 static void DrawSilk (int side, const BoxType *);
 static int via_callback (const BoxType * b, void *cl);
@@ -704,6 +705,26 @@ DrawMask (int side, BoxType * screen)
     }
 }
 
+/* ---------------------------------------------------------------------------
+ * draws solder paste layer for a given side of the board
+ */
+static void
+DrawPaste (int side, BoxType *drawn_area)
+{
+  gui->set_color (Output.fgGC, PCB->ElementColor);
+  ALLPAD_LOOP (PCB->Data);
+  {
+    if (ON_SIDE (pad, side) && !TEST_FLAG (NOPASTEFLAG, pad) && pad->Mask > 0)
+      {
+        if (pad->Mask < pad->Thickness)
+          DrawPadLowLevel (Output.fgGC, pad, true, true);
+        else
+          DrawPadLowLevel (Output.fgGC, pad, false, false);
+      }
+  }
+  ENDALL_LOOP;
+}
+
 static void
 DrawRats (BoxTypePtr drawn_area)
 {
